registry/config.go

```diff
@@ -328,35 +271,10 @@ func (config *ServiceConfig) NewRepositoryInfo(reposName reference.Named) (*Repo
 		return nil, err
 	}
 
-	if repoInfo.Index.Official {
-		repoInfo.LocalName, err = normalizeLibraryRepoName(repoInfo.RemoteName)
-		if err != nil {
-			return nil, err
-		}
-		repoInfo.RemoteName = repoInfo.LocalName
-
-		// If the normalized name does not contain a '/' (e.g. "foo")
-		// then it is an official repo.
-		if strings.IndexRune(repoInfo.RemoteName.Name(), '/') == -1 {
-			repoInfo.Official = true
-			// Fix up remote name for official repos.
-			repoInfo.RemoteName, err = reference.WithName("library/" + repoInfo.RemoteName.Name())
-			if err != nil {
-				return nil, err
-			}
-		}
+	repoInfo.CanonicalName = reference.WithHostname(reposName)
+	repoInfo.LocalName = reference.Normalize(reposName)
```

Comment:
I guess it would be better to turn this question around at all and ask why `ParseNamed` normalizes at all if many other places can't count on this normalization happening automatically.